runtime/interpreter/interpreter_expression.go

```diff
@@ -1436,9 +1428,6 @@ func (interpreter *Interpreter) VisitAttachExpression(attachExpression *ast.Atta
 		panic(errors.NewUnreachableError())
 	}
 
-	// Because `self` in attachments is a reference, we need to track the attachment if it's a resource
-	interpreter.trackReferencedResourceKindedValue(attachment.StorageID(), attachment)
```

Comment:
This is the only place where the reference is tracked outside of creating a reference.
Removing this didn't cause any tests to fail. So I assume this is redundant. 
But lemme know if you can think of a way to exploit this

Reply:
This is the one I removed in https://github.com/onflow/cadence/pull/2914 because it was redundant, so I'm pretty sure it's safe to delete here.